File: src/main/scala/com/github/mauricio/postgresql/messages/backend/RowDescriptionMessage.scala
package com.github.mauricio.postgresql.messages.backend

/**
 * User: mauricio
 * Date: 3/31/13
 * Time: 1:15 AM
 */
case class RowDescriptionMessage ( val columnDatas : Array[ColumnData] )
  extends Message( Message.RowDescription )

File: postgresql-async/src/main/java/com/github/mauricio/async/db/postgresql/messages/backend/RowDescriptionMessage.kt
package com.github.mauricio.async.db.postgresql.messages.backend

data class RowDescriptionMessage(val columnDatas: List<PostgreSQLColumnData>) : ServerMessage(ServerMessage.RowDescription) {
  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (javaClass != other?.javaClass) return false

    other as RowDescriptionMessage

    return columnDatas == other.columnDatas
  }

  override fun hashCode(): Int {
    return columnDatas.hashCode()
  }
}
